### PR TITLE
add matrix_inversion example

### PR DESCRIPTION
**Pull request by miguelraz**
I heard y'all like simd examples in #110

### CODE REVIEW

File: crates/core_simd/examples/matrix_inversion.rs [hunk 1]
```diff
@@ -0,0 +1,319 @@
+//! 4x4 matrix inverse
+// Code ported from the `packed_simd` crate
+// Run this code with `cargo test --example matrix_inversion`
+#![feature(custom_inner_attributes)]
+#![deny(rust_2018_idioms)]
+#![allow(clippy::must_use_candidate)]
+#![feature(array_chunks)]
```
Comment (scottmcm) [inline crates/core_simd/examples/matrix_inversion.rs:7]: Ctrl-F can't actually find any call to this?

Comment (miguelraz) [reply crates/core_simd/examples/matrix_inversion.rs:7]: Oh neat, I only need the `array_chunks`. Thanks.

**Comment by miguelraz** [comment]
Ready to merge @workingjubilee .

File: crates/core_simd/examples/matrix_inversion.rs [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+pub struct Matrix4x4([[f32; 4]; 4]);
+
+#[allow(clippy::too_many_lines)]
+pub fn scalar_inv4x4(m: Matrix4x4) -> Option<Matrix4x4> {
+    let m = m.0;
+
+    let mut inv = [
+        [ // row 0:
+            // 0,0:
+            m[1][1]  * m[2][2] * m[3][3] -
```
Comment (calebzulawski) [inline crates/core_simd/examples/matrix_inversion.rs:18]: Some of these lines seem to have extra spaces for some reason?

**Review by calebzulawski — CHANGES_REQUESTED** [review]

File: crates/core_simd/examples/matrix_inversion.rs (hunk 2 above)
Comment (workingjubilee) [reply crates/core_simd/examples/matrix_inversion.rs:18]: Yeah, `cargo fmt` actually is not _totally_ awful formatting for this, surprisingly, so I suggest that as a base.

Comment (miguelraz) [reply crates/core_simd/examples/matrix_inversion.rs:18]: I think the function has the formatter turned off.

**Review by workingjubilee — APPROVED** [review]

File: crates/core_simd/examples/matrix_inversion.rs (hunk 2 above)
Comment (workingjubilee) [reply crates/core_simd/examples/matrix_inversion.rs:18]: We can address formatting concerns for the repo later.

**Pull request merged**